src/value.h

```diff
@@ -78,7 +82,9 @@ void SIValue_Free(SIValue *v);
 void SIValueVector_Append(SIValueVector *v, SIValue val);
 void SIValueVector_Free(SIValueVector *v);
 
-SIValue SI_StringVal(const char *s);
+SIValue SI_DuplicateStringVal(const char *s);
+SIValue SI_TransferStringVal(char *s);
```

Comment:
What's the purpose of this function, I don't think we're using it.

Reply:
I had the idea that heap-allocated strings which eventually must be freed, but are at some point being added to an SIValue (such as a string from the parser) could use this function to transfer their ownership and get freed without an unnecessary alloc.

It doesn't work in the parser example because those strings are malloced, rather than rm_malloced (RediSearch actually keeps track of allocation source to handle issues like this).

I'll delete it if it doesn't end up used anywhere!